src/org/labkey/test/components/ui/grids/TabbedGridPanel.java

```diff
@@ -0,0 +1,126 @@
+package org.labkey.test.components.ui.grids;
+
+import org.labkey.test.Locator;
+import org.labkey.test.components.Component;
+import org.labkey.test.components.WebDriverComponent;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+
+import java.util.List;
+
+import static org.labkey.test.WebDriverWrapper.WAIT_FOR_JAVASCRIPT;
+
+/**
+ * TabbedGridPanel wraps components/src/public/QueryModel/TabbedGridPanel.tsx
+ * The model of a tabbed grid panel is different from previous models; rather than being a component of the
+ * QueryGrid, the TabbedGridPanel contains multiple grids that are shown/hidden via bootstrap style
+ */
+public class TabbedGridPanel extends WebDriverComponent<TabbedGridPanel.ElementCache>
+{
+    private final WebElement _el;
+    private final WebDriver _driver;
+
+    protected TabbedGridPanel(WebElement element, WebDriver driver)
+    {
+        _el = element;
+        _driver = driver;
+    }
+
+    @Override
+    public WebElement getComponentElement()
+    {
+        return _el;
+    }
+
+    @Override
+    public WebDriver getDriver()
+    {
+        return _driver;
+    }
+
+    public List<String> getTabs()
+    {
+        return getWrapper().getTexts(elementCache().navTabs());
+    }
+
+    private boolean isSelected(String tabText)
+    {
+        String tabClass = elementCache().navTab(tabText).getAttribute("class");
+        return "active".equals(tabClass);
+    }
+
+    public QueryGrid selectGrid(String tabText)
+    {
+        if (!isSelected(tabText))
+        {
+            var tab = elementCache().navTab(tabText);
+            getWrapper().shortWait().until(ExpectedConditions.elementToBeClickable(tab));
+            tab.click();
+            getWrapper().waitFor(()-> isSelected(tabText), "tab did not become selected in time", 2000);
+        }
+
+        return getSelectedGrid();
+    }
+
+    public QueryGrid getSelectedGrid()
+    {
+        return new QueryGrid.QueryGridFinder(getDriver()).waitFor(this);
+    }
```

Comment:
If the class documentation is accurate and the grids are only shown and hidden when switching tabs, I would expect this to always return the first grid regardless of which tab is selected. I don't see anything in `QueryGrid.QueryGridFinder` that would restrict this to finding only the visible grid.

Reply:
It turns out that the comment isn't accurate, the grid component switching appears to be handled as virtual-dom actions.